src/group_theory/submonoid/membership.lean

```diff
@@ -220,7 +220,7 @@ lemma powers_subset {n : M} {P : submonoid M} (h : n ∈ P) : powers n ≤ P :=
 λ x hx, match x, hx with _, ⟨i, rfl⟩ := P.pow_mem h i end
 
 /-- Exponentiation map from natural numbers to powers. -/
-def pow (n : M) (m : ℕ) : powers n := ⟨n ^ m, m, rfl⟩
+def pow (n : M) (m : ℕ) : powers n := (powers_hom M n).mrange_restrict m
```

Comment:
Strictly this should be
```suggestion
def pow (n : M) (m : ℕ) : powers n :=
(powers_hom M n).mrange_restrict (multiplicative.of_add m)
```

Reply:
This threw some errors downstream so I reverted back to `(powers_hom M n).mrange_restrict m`.

Reply:
What were the errors? That indicates that we might have something wrong somewhere else too

Reply:
See my comments below

Reply:
I think the errors were coming from the fact that I hadn't added `to_add` to `pow` and `log` in the `mul_equiv` definitions. It works now after adding those.